packages/framework/src/Component/Redis/RedisVersionsFacade.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Component\Redis;
+
+use Redis;
+
+class RedisVersionsFacade
+{
+    /**
+     * @var \Redis
+     */
+    protected $globalClient;
+
+    /**
+     * @var string
+     */
+    protected $currentVersion;
+
+    /**
+     * @param \Redis $globalClient
+     * @param string $currentVersion
+     */
+    public function __construct(Redis $globalClient, string $currentVersion)
+    {
+        $this->globalClient = $globalClient;
+        $this->currentVersion = $currentVersion;
+    }
+
+    public function cleanOldCache(): void
+    {
+        $versionPattern = '[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]*'; // redis pattern is a glob style and doesn't support repetitions
+
+        $all = $this->globalClient->keys($versionPattern);
+        $toRemove = [];
+
+        $prefix = (string)$this->globalClient->getOption(Redis::OPT_PREFIX);
+        $currentVersionPrefix = $prefix . $this->currentVersion;
+        foreach ($all as $key) {
```

Comment:
Probably that may be a problem for really big e-shops. For example, if a million records are to be deleted that foreach will be very slow.

Reply:
https://redis.io/commands/unlink

Reply:
You're right. `Keys` is a performance killer.

Reply:
I rewrote it to scan the keys, and it seems fine. I was able to delete about 1M of records in a minute.